docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,527 @@ overview page.
+      rules: <RULES DEFINITION>
+
+There is some recording and alert rules defined in the `Loki repository`_ that
+could be used as a base, then we could enrich these rules later when we will
+have better operational knowledge.
+
+.. _Loki repository: https://github.com/grafana/loki/tree/master/production/loki-mixin
+
+Grafana
+"""""""
+
+To be able to query logs from Loki, we need to add a Grafana datasource,
+this is done adding a ConfigMap ``loki-grafana-datasource`` in
+``metalk8s-monitoring`` Namespace as follows:
+
+.. code-block:: yaml
+
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: loki-grafana-datasource
+    namespace: metalk8s-monitoring
+    labels:
+      grafana_datasource: "1"
+  data:
+    loki-datasource.yaml: |-
+      apiVersion: 1
+      datasources:
+      - name: Loki
+        type: loki
+        access: proxy
+        url: http://loki.loki.svc.cluster.local:3100
+        version: 1
+
+To display the logs we also need a dashboard, adding a ConfigMap
+``loki-logs-dashboard`` in Namespace ``metalk8s-monitoring``:
+
+.. code-block:: yaml
+
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: loki-logs-dashboard
+    namespace: metalk8s-monitoring
+    labels:
+      grafana_dashboard: "1"
+  data:
+    loki-logs.json: <DASHBOARD DEFINITION>
+
+For the dashboard we will use a view with a simple log panel and variables
+representing labels to filter on.
+Since journal and kubernetes logs will not have the same labels, we could
+either have 2 distinct dashboards or 2 log panels in the same.
+An example of what we want is `Loki dashboard`_.
+
+.. _Loki dashboard: https://github.com/grafana/loki/blob/master/production/loki-mixin/dashboard-loki-logs.json
+
+.. note::
+
+  The ``grafana_datasource: "1"`` and ``grafana_dashboard: "1"`` labels are
+  what is used by the Prometheus Operator to retrieve datasource and dashboard
+  for Grafana.
+
+Loki Volume Purge
+~~~~~~~~~~~~~~~~~
+
+Even with a max retention period, the logs could grow faster than what was
+expected and fill up the volume.
+Since there is no retention based on size in Loki yet, we need to add some
+specific monitoring (with prediction on volume usage) and alerting to ensure
+that an administrator will be warned if such a case would happen.
+The alert message should be clear and provide an URL to a run book to help
+the administrator resolving the issue.
+
+To fix this issue, the administrator should purge oldest log chunk files
+from Loki volume, which can be achieved by connecting to the pod and manually
+removing them.
+If the growth of logs is not something transient, the administrator should
+also be advised to lower the retention period or replace the Loki volume
+by a bigger one.
+
+Iterations
+~~~~~~~~~~
+
+Iteration 1
+"""""""""""
+
+The goal is to have a working log centralization system, with logs accessible
+from Grafana:
+
+* Deploy Fluent Bit and Loki
+* Deploy Grafana datasource & dashboard
+* Deploy Loki volumes purge mechanism
```

Comment:
? Which mechanism?

Reply:
No mechanism now, it's a leftover.
It's more about having dedicated alert rules for Loki, I'll update this.